src/NuGet.Server.Core/Infrastructure/ServerPackageRepository.cs

```diff
@@ -108,6 +108,35 @@ public class ServerPackageRepository
 
         private string CacheFileName => _settingsProvider.GetStringSetting("cacheFileName", null);
 
+        private TimeSpan InitialCacheRebuildAfter
+        {
+            get
+            {
+                const int defaultValue = 15;
+                var value = _settingsProvider.GetIntSetting("initialCacheRebuildAfterSeconds", defaultValue);
+                if (value <= 0)
+                {
+                    value = defaultValue;
+                }
+
+                return TimeSpan.FromSeconds(value);
+            }
+        }
+        private TimeSpan CacheRebuildFrequency
+        {
+            get
+            {
+                const int defaultValue = 60;
+                var value = _settingsProvider.GetIntSetting("cacheRebuildFrequencyInMinutes", defaultValue);
+                if (value <= 0)
+                {
+                    value = defaultValue;
+                }
```

Comment:
Put the 5 lines above into its own function, so they are not copy-pasted maybe?

Reply:
Good idea!